2025-01-04 17:39:50 - Server listening on ('0.0.0.0', 3102)
2025-01-04 17:39:51 - Public server ip: 46.121.5.143, local server ip: 192.168.56.1
2025-01-04 17:39:51 - Main thread: before accepting ...
2025-01-05 21:37:20 - Server listening on ('0.0.0.0', 3102)
2025-01-05 21:37:20 - Public server ip: 46.121.5.143, local server ip: 192.168.56.1
2025-01-05 21:37:20 - Main thread: before accepting ...
2025-01-05 21:37:22 - New Client number 1 from ('192.168.56.1', 54329)
2025-01-05 21:39:03 - Client 1 disconnected
2025-01-05 21:39:03 - Client 1 Exit
2025-01-05 21:39:05 - New Client number 2 from ('192.168.56.1', 54351)
2025-01-05 21:39:21 - Client 2 disconnected
2025-01-05 21:39:21 - Client 2 Exit
2025-01-05 21:39:23 - New Client number 3 from ('192.168.56.1', 54363)
2025-01-05 21:39:58 - Client 3 disconnected
2025-01-05 21:39:58 - Client 3 Exit
2025-01-05 21:40:00 - New Client number 4 from ('192.168.56.1', 54385)
2025-01-05 21:40:22 - Client 4 disconnected
2025-01-05 21:40:22 - Client 4 Exit
2025-01-05 21:40:25 - New Client number 5 from ('192.168.56.1', 54400)
2025-01-05 21:40:48 - Client 5 disconnected
2025-01-05 21:40:48 - Client 5 Exit
2025-01-05 21:40:50 - New Client number 6 from ('192.168.56.1', 54418)
2025-01-05 21:41:50 - Client 6 disconnected
2025-01-05 21:41:50 - Client 6 Exit
2025-01-05 21:41:53 - New Client number 7 from ('192.168.56.1', 54437)
2025-01-05 21:49:56 - Client 7 Exit
2025-01-05 21:49:56 - Client 7 disconnected
2025-01-05 21:50:02 - New Client number 8 from ('192.168.56.1', 54543)
2025-01-05 21:53:42 - Client 8 disconnected
2025-01-05 21:53:42 - Client 8 Exit
2025-01-05 21:53:44 - New Client number 9 from ('192.168.56.1', 54566)
2025-01-05 21:56:51 - Client 9 disconnected
2025-01-05 21:56:51 - Client 9 Exit
2025-01-05 21:56:53 - New Client number 10 from ('192.168.56.1', 54597)
2025-01-05 22:00:31 - Client 10 disconnected
2025-01-05 22:00:31 - Client 10 Exit
2025-01-05 22:00:33 - New Client number 11 from ('192.168.56.1', 54631)
2025-01-05 22:01:20 - Client 11 disconnected
2025-01-05 22:01:20 - Client 11 Exit
2025-01-05 22:01:22 - New Client number 12 from ('192.168.56.1', 54653)
2025-01-05 22:01:33 - Client 12 disconnected
2025-01-05 22:01:33 - Client 12 Exit
2025-01-05 22:01:35 - New Client number 13 from ('192.168.56.1', 54662)
2025-01-05 22:01:51 - Client 13 disconnected
2025-01-05 22:01:51 - Client 13 Exit
2025-01-05 22:01:54 - New Client number 14 from ('192.168.56.1', 54680)
2025-01-05 22:02:53 - Client 14 disconnected
2025-01-05 22:02:53 - Client 14 Exit
2025-01-05 22:02:56 - New Client number 15 from ('192.168.56.1', 54695)
2025-01-05 22:25:16 - Server listening on ('0.0.0.0', 3102)
2025-01-05 22:25:17 - Public server ip: 46.121.5.143, local server ip: 192.168.56.1
2025-01-05 22:25:17 - Main thread: before accepting ...
2025-01-05 22:25:19 - New Client number 1 from ('192.168.56.1', 55749)
2025-01-05 22:26:44 - Client 1 disconnected
2025-01-05 22:26:44 - Client 1 Exit
2025-01-05 22:26:47 - New Client number 2 from ('192.168.56.1', 55777)
2025-01-05 22:31:47 - Server listening on ('0.0.0.0', 3102)
2025-01-05 22:31:48 - Public server ip: 46.121.5.143, local server ip: 192.168.56.1
2025-01-05 22:31:48 - Main thread: before accepting ...
2025-01-05 22:31:50 - New Client number 1 from ('192.168.56.1', 55868)
2025-01-05 22:31:50 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 838, in handle_request
    to_send = protocol_build_reply(request, tid, sock)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 529, in protocol_build_reply
    if amount > len(items): amount = len(items)
       ^^^^^^^^^^^^^^^^^^^
TypeError: '>' not supported between instances of 'str' and 'int'
2025-01-05 22:31:50 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 838, in handle_request
    to_send = protocol_build_reply(request, tid, sock)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 528, in protocol_build_reply
    if (code == "GEDD" or code == "GESD" or code == "GETD") and amount - prev_amount <= 0: amount = 0
                                                                ~~~~~~~^~~~~~~~~~~~~
TypeError: unsupported operand type(s) for -: 'str' and 'int'
2025-01-05 22:32:06 - Server listening on ('0.0.0.0', 3102)
2025-01-05 22:32:07 - Public server ip: 46.121.5.143, local server ip: 192.168.56.1
2025-01-05 22:32:07 - Main thread: before accepting ...
2025-01-05 22:32:08 - New Client number 1 from ('192.168.56.1', 55888)
2025-01-05 22:32:11 - Client 1 disconnected
2025-01-05 22:32:11 - Client 1 Exit
2025-01-05 22:32:13 - New Client number 2 from ('192.168.56.1', 55898)
2025-01-05 22:33:07 - Client 2 disconnected
2025-01-05 22:33:07 - Client 2 Exit
2025-01-05 22:33:10 - New Client number 3 from ('192.168.56.1', 55919)
2025-01-05 22:34:14 - Server listening on ('0.0.0.0', 3102)
2025-01-05 22:34:14 - Public server ip: 46.121.5.143, local server ip: 192.168.56.1
2025-01-05 22:34:15 - Main thread: before accepting ...
2025-01-05 22:34:17 - New Client number 1 from ('192.168.56.1', 55956)
2025-01-05 22:35:06 - Client 1 disconnected
2025-01-05 22:35:06 - Client 1 Exit
2025-01-05 22:35:09 - New Client number 2 from ('192.168.56.1', 55981)
2025-01-05 22:40:15 - Client 2 disconnected
2025-01-05 22:40:15 - Client 2 Exit
2025-01-05 22:40:17 - New Client number 3 from ('192.168.56.1', 56051)
2025-01-05 22:40:49 - Client 3 disconnected
2025-01-05 22:40:49 - Client 3 Exit
2025-01-05 22:40:52 - New Client number 4 from ('192.168.56.1', 56067)
2025-01-05 22:52:11 - Client 4 disconnected
2025-01-05 22:52:11 - Client 4 Exit
2025-01-05 22:52:14 - Server listening on ('0.0.0.0', 3102)
2025-01-05 22:52:15 - Public server ip: 46.121.5.143, local server ip: 192.168.56.1
2025-01-05 22:52:15 - Main thread: before accepting ...
2025-01-05 22:52:19 - New Client number 1 from ('192.168.56.1', 56201)
2025-01-05 22:52:19 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 870, in handle_request
    to_send = protocol_build_reply(request, tid, sock)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 510, in protocol_build_reply
    sort = fields[3]
           ~~~~~~^^^
IndexError: list index out of range
2025-01-05 22:52:19 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 870, in handle_request
    to_send = protocol_build_reply(request, tid, sock)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 510, in protocol_build_reply
    sort = fields[3]
           ~~~~~~^^^
IndexError: list index out of range
2025-01-05 22:52:45 - Client 1 disconnected
2025-01-05 22:52:45 - Client 1 Exit
2025-01-05 22:52:48 - New Client number 2 from ('192.168.56.1', 56217)
2025-01-05 22:52:53 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 870, in handle_request
    to_send = protocol_build_reply(request, tid, sock)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 548, in protocol_build_reply
    items = sorted(items, key=lambda x: str_to_date(x.split("~")[1]), reverse=sort_direction)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 548, in <lambda>
    items = sorted(items, key=lambda x: str_to_date(x.split("~")[1]), reverse=sort_direction)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 183, in str_to_date
    return datetime.strptime(str, format)
           ^^^^^^^^
NameError: name 'datetime' is not defined. Did you forget to import 'datetime'
2025-01-05 22:52:53 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 870, in handle_request
    to_send = protocol_build_reply(request, tid, sock)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 546, in protocol_build_reply
    items = sorted(items, key=lambda x: str_to_date(x.split("~")[2]), reverse=sort_direction)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 546, in <lambda>
    items = sorted(items, key=lambda x: str_to_date(x.split("~")[2]), reverse=sort_direction)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 183, in str_to_date
    return datetime.strptime(str, format)
           ^^^^^^^^
NameError: name 'datetime' is not defined. Did you forget to import 'datetime'
2025-01-05 22:53:08 - Server listening on ('0.0.0.0', 3102)
2025-01-05 22:53:08 - Public server ip: 46.121.5.143, local server ip: 192.168.56.1
2025-01-05 22:53:08 - Main thread: before accepting ...
2025-01-05 22:53:09 - New Client number 1 from ('192.168.56.1', 56243)
2025-01-05 22:53:12 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 870, in handle_request
    to_send = protocol_build_reply(request, tid, sock)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 548, in protocol_build_reply
    items = sorted(items, key=lambda x: str_to_date(x.split("~")[1]), reverse=sort_direction)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 548, in <lambda>
    items = sorted(items, key=lambda x: str_to_date(x.split("~")[1]), reverse=sort_direction)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 183, in str_to_date
    return datetime.strptime(str, format)
           ^^^^^^^^^^^^^^^^^
AttributeError: module 'datetime' has no attribute 'strptime'
2025-01-05 22:53:12 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 870, in handle_request
    to_send = protocol_build_reply(request, tid, sock)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 546, in protocol_build_reply
    items = sorted(items, key=lambda x: str_to_date(x.split("~")[2]), reverse=sort_direction)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 546, in <lambda>
    items = sorted(items, key=lambda x: str_to_date(x.split("~")[2]), reverse=sort_direction)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\server.py", line 183, in str_to_date
    return datetime.strptime(str, format)
           ^^^^^^^^^^^^^^^^^
AttributeError: module 'datetime' has no attribute 'strptime'
2025-01-05 22:53:31 - Server listening on ('0.0.0.0', 3102)
2025-01-05 22:53:31 - Public server ip: 46.121.5.143, local server ip: 192.168.56.1
2025-01-05 22:53:31 - Main thread: before accepting ...
2025-01-05 22:53:32 - New Client number 1 from ('192.168.56.1', 56274)
2025-01-05 22:55:01 - Client 1 disconnected
2025-01-05 22:55:01 - Client 1 Exit
2025-01-05 22:55:03 - New Client number 2 from ('192.168.56.1', 56297)
2025-01-05 22:55:53 - Server listening on ('0.0.0.0', 3102)
2025-01-05 22:55:53 - Public server ip: 46.121.5.143, local server ip: 192.168.56.1
2025-01-05 22:55:53 - Main thread: before accepting ...
2025-01-05 22:55:55 - New Client number 1 from ('192.168.56.1', 56339)
2025-01-05 22:57:50 - Server listening on ('0.0.0.0', 3102)
2025-01-05 22:57:51 - Public server ip: 46.121.5.143, local server ip: 192.168.56.1
2025-01-05 22:57:51 - Main thread: before accepting ...
2025-01-05 22:58:27 - New Client number 1 from ('192.168.56.1', 56383)
2025-01-05 22:58:57 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\40b3f2df4be9426db5e083ae79b08c39.lock'
2025-01-05 22:58:58 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\15ec7e434f854785a8871017d4731943.lock'
2025-01-05 22:58:59 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\15ec7e434f854785a8871017d4731943.lock'
2025-01-05 22:59:01 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\15ec7e434f854785a8871017d4731943.lock'
2025-01-05 22:59:02 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\15ec7e434f854785a8871017d4731943.lock'
2025-01-05 22:59:03 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\15ec7e434f854785a8871017d4731943.lock'
2025-01-05 23:01:36 - Client 1 disconnected
2025-01-05 23:01:36 - Client 1 Exit
2025-01-05 23:01:38 - New Client number 2 from ('192.168.56.1', 56424)
2025-01-05 23:02:19 - Client 2 disconnected
2025-01-05 23:02:19 - Client 2 Exit
2025-01-05 23:02:21 - New Client number 3 from ('192.168.56.1', 56442)
2025-01-05 23:09:12 - Client 3 disconnected
2025-01-05 23:09:12 - Client 3 Exit
2025-01-05 23:09:15 - Server listening on ('0.0.0.0', 3102)
2025-01-05 23:09:16 - Public server ip: 46.121.5.143, local server ip: 192.168.56.1
2025-01-05 23:09:16 - Main thread: before accepting ...
2025-01-05 23:09:20 - New Client number 1 from ('192.168.56.1', 56566)
2025-01-05 23:09:51 - Client 1 disconnected
2025-01-05 23:09:51 - Client 1 Exit
2025-01-05 23:09:53 - New Client number 2 from ('192.168.56.1', 56580)
2025-01-05 23:17:45 - Client 2 disconnected
2025-01-05 23:17:45 - Client 2 Exit
2025-01-05 23:17:48 - Server listening on ('0.0.0.0', 3102)
2025-01-05 23:17:48 - Public server ip: 46.121.5.143, local server ip: 192.168.56.1
2025-01-05 23:17:49 - Main thread: before accepting ...
2025-01-05 23:17:52 - New Client number 1 from ('192.168.56.1', 56690)
2025-01-05 23:18:02 - Client 1 disconnected
2025-01-05 23:18:02 - Client 1 Exit
2025-01-05 23:18:04 - New Client number 2 from ('192.168.56.1', 56700)
2025-01-05 23:18:22 - Client 2 disconnected
2025-01-05 23:18:22 - Client 2 Exit
2025-01-05 23:18:25 - New Client number 3 from ('192.168.56.1', 56716)
2025-01-05 23:18:46 - Client 3 disconnected
2025-01-05 23:18:46 - Client 3 Exit
2025-01-05 23:18:48 - New Client number 4 from ('192.168.56.1', 56736)
2025-01-05 23:19:13 - Client 4 disconnected
2025-01-05 23:19:13 - Client 4 Exit
2025-01-05 23:19:15 - New Client number 5 from ('192.168.56.1', 56749)
2025-01-05 23:19:51 - Client 5 disconnected
2025-01-05 23:19:51 - Client 5 Exit
2025-01-05 23:19:53 - New Client number 6 from ('192.168.56.1', 56768)
2025-01-05 23:20:04 - Client 6 disconnected
2025-01-05 23:20:04 - Client 6 Exit
2025-01-05 23:20:06 - New Client number 7 from ('192.168.56.1', 56781)
2025-01-05 23:21:57 - Client 7 disconnected
2025-01-05 23:21:57 - Client 7 Exit
2025-01-05 23:21:59 - New Client number 8 from ('192.168.56.1', 56813)
2025-01-05 23:22:42 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\0ee2fcd7b4df407aae67900d78444129'
2025-01-05 23:22:42 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\0ee5566881554c19bd86cae20b64e0c4'
2025-01-05 23:22:42 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\15ec7e434f854785a8871017d4731943'
2025-01-05 23:22:42 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\197d85d6b74a4ed4aaa30e5c89aea9cf'
2025-01-05 23:22:42 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\1a3ed7f989d84be59cd9470c72f6ae53'
2025-01-05 23:22:42 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\244d0547564848e1822acdc03e141c4a'
2025-01-05 23:22:42 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\5988896d687f4d3ebe3c2352b34586d5'
2025-01-05 23:22:42 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\6448cd7a6bb848dd81fd6d02aed7c2c2'
2025-01-05 23:22:42 - Traceback (most recent call last):
  File "c:\Users\idanh\Desktop\Cyber\My Projects\Final Project\Main\server\modules\client_requests.py", line 587, in clean_db
    os.remove(server_path + "\\cloud\\" + name)
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'c:\\Users\\idanh\\Desktop\\Cyber\\My Projects\\Final Project\\Main\\server\\cloud\\bad767c56f3f49119a79a5711773e404'
2025-01-05 23:23:03 - Server listening on ('0.0.0.0', 3102)
2025-01-05 23:23:03 - Public server ip: 46.121.5.143, local server ip: 192.168.56.1
2025-01-05 23:23:04 - Main thread: before accepting ...
2025-01-05 23:23:05 - New Client number 1 from ('192.168.56.1', 56840)
2025-01-05 23:23:32 - Client 1 disconnected
2025-01-05 23:23:32 - Client 1 disconnected
2025-01-05 23:23:32 - Client 1 Exit
2025-01-05 23:54:25 - Server listening on ('0.0.0.0', 3102)
2025-01-05 23:54:26 - Public server ip: 46.121.5.143, local server ip: 192.168.1.247
2025-01-05 23:54:28 - Main thread: before accepting ...
2025-01-05 23:54:42 - New Client number 1 from ('192.168.1.247', 56020)
2025-01-05 23:56:30 - Client 1 disconnected
2025-01-05 23:56:30 - Client 1 Exit
2025-01-05 23:56:35 - New Client number 2 from ('192.168.1.247', 56065)
2025-01-05 23:57:05 - Client 2 disconnected
2025-01-05 23:57:05 - Client 2 Exit
2025-01-05 23:57:09 - New Client number 3 from ('192.168.1.247', 56077)
2025-01-06 00:02:33 - Client 3 disconnected
2025-01-06 00:02:33 - Client 3 Exit
2025-01-06 00:02:43 - New Client number 4 from ('192.168.1.247', 56200)
2025-01-06 00:03:30 - Client 4 disconnected
2025-01-06 00:03:30 - Client 4 Exit
2025-01-06 00:03:35 - New Client number 5 from ('192.168.1.247', 56212)
2025-01-06 00:04:37 - Client 5 disconnected
2025-01-06 00:04:37 - Client 5 Exit
2025-01-06 00:04:41 - New Client number 6 from ('192.168.1.247', 56226)
2025-01-06 00:07:15 - Client 6 disconnected
2025-01-06 00:07:15 - Client 6 Exit
2025-01-06 00:07:20 - New Client number 7 from ('192.168.1.247', 56252)
2025-01-06 00:07:37 - Client 7 disconnected
2025-01-06 00:07:37 - Client 7 Exit
2025-01-06 00:07:42 - New Client number 8 from ('192.168.1.247', 56265)
2025-01-06 00:08:08 - Client 8 disconnected
2025-01-06 00:08:08 - Client 8 Exit
2025-01-06 00:08:12 - New Client number 9 from ('192.168.1.247', 56277)
2025-01-06 00:09:30 - Client 9 disconnected
2025-01-06 00:09:30 - Client 9 Exit
2025-01-06 00:09:34 - New Client number 10 from ('192.168.1.247', 56299)
2025-01-06 00:10:27 - Client 10 disconnected
2025-01-06 00:10:27 - Client 10 Exit
2025-01-06 00:10:31 - New Client number 11 from ('192.168.1.247', 56320)
2025-01-06 00:12:36 - Client 11 disconnected
2025-01-06 00:12:36 - Client 11 Exit
2025-01-06 00:12:40 - New Client number 12 from ('192.168.1.247', 56358)
2025-01-06 00:16:25 - Client 12 disconnected
2025-01-06 00:16:25 - Client 12 Exit
2025-01-06 00:16:29 - New Client number 13 from ('192.168.1.247', 56401)
2025-01-06 00:16:59 - Client 13 disconnected
2025-01-06 00:16:59 - Client 13 Exit
2025-01-06 00:17:03 - New Client number 14 from ('192.168.1.247', 56419)
2025-01-06 00:18:59 - Client 14 disconnected
2025-01-06 00:18:59 - Client 14 Exit
2025-01-06 00:19:03 - New Client number 15 from ('192.168.1.247', 56460)
2025-01-06 00:20:14 - Client 15 disconnected
2025-01-06 00:20:14 - Client 15 Exit
2025-01-06 00:20:19 - New Client number 16 from ('192.168.1.247', 56493)
2025-01-06 00:21:39 - Client 16 disconnected
2025-01-06 00:21:39 - Client 16 Exit
2025-01-06 00:21:44 - New Client number 17 from ('192.168.1.247', 56534)
2025-01-06 00:22:18 - Client 17 disconnected
2025-01-06 00:22:18 - Client 17 Exit
2025-01-06 00:22:22 - New Client number 18 from ('192.168.1.247', 56558)
2025-01-06 00:22:53 - Client 18 disconnected
2025-01-06 00:22:53 - Client 18 Exit
2025-01-06 00:22:57 - New Client number 19 from ('192.168.1.247', 56576)
2025-01-06 00:23:10 - Client 19 disconnected
2025-01-06 00:23:10 - Client 19 Exit
2025-01-06 00:23:15 - New Client number 20 from ('192.168.1.247', 56590)
2025-01-06 00:23:44 - Client 20 disconnected
2025-01-06 00:23:44 - Client 20 Exit
2025-01-06 00:23:48 - New Client number 21 from ('192.168.1.247', 56606)
2025-01-06 00:26:31 - Client 21 disconnected
2025-01-06 00:26:31 - Client 21 Exit
2025-01-06 00:26:35 - New Client number 22 from ('192.168.1.247', 56659)
2025-01-06 00:27:28 - Client 22 disconnected
2025-01-06 00:27:28 - Client 22 Exit
2025-01-06 00:27:33 - New Client number 23 from ('192.168.1.247', 56683)
2025-01-06 00:27:46 - Client 23 disconnected
2025-01-06 00:27:46 - Client 23 Exit
2025-01-06 00:27:50 - New Client number 24 from ('192.168.1.247', 56694)
2025-01-06 00:29:22 - Client 24 disconnected
2025-01-06 00:29:22 - Client 24 Exit
2025-01-06 00:29:26 - New Client number 25 from ('192.168.1.247', 56719)
2025-01-06 00:30:34 - Client 25 disconnected
2025-01-06 00:30:34 - Client 25 Exit
2025-01-06 00:30:38 - New Client number 26 from ('192.168.1.247', 56738)
2025-01-06 00:32:35 - Client 26 disconnected
2025-01-06 00:32:35 - Client 26 Exit
2025-01-06 00:32:39 - New Client number 27 from ('192.168.1.247', 56769)
2025-01-06 00:33:13 - Client 27 disconnected
2025-01-06 00:33:13 - Client 27 Exit
2025-01-06 00:33:18 - New Client number 28 from ('192.168.1.247', 56787)
2025-01-06 00:33:37 - Client 28 disconnected
2025-01-06 00:33:37 - Client 28 Exit
2025-01-06 00:33:41 - New Client number 29 from ('192.168.1.247', 56801)
2025-01-06 00:33:59 - Client 29 disconnected
2025-01-06 00:33:59 - Client 29 Exit
2025-01-06 00:34:04 - New Client number 30 from ('192.168.1.247', 56816)
2025-01-06 00:34:12 - Client 30 disconnected
2025-01-06 00:34:12 - Client 30 Exit
2025-01-06 00:34:41 - New Client number 31 from ('192.168.1.247', 56844)
2025-01-06 00:35:09 - Client 31 disconnected
2025-01-06 00:35:09 - Client 31 Exit
2025-01-06 00:35:13 - New Client number 32 from ('192.168.1.247', 56862)
2025-01-06 00:40:22 - Client 32 disconnected
2025-01-06 00:40:22 - Client 32 Exit
2025-01-06 00:40:26 - New Client number 33 from ('192.168.1.247', 56926)
2025-01-06 00:40:34 - Client 33 disconnected
2025-01-06 00:40:34 - Client 33 Exit
2025-01-06 00:40:38 - New Client number 34 from ('192.168.1.247', 56938)
2025-01-06 00:42:23 - Client 34 disconnected
2025-01-06 00:42:23 - Client 34 Exit
2025-01-06 00:42:28 - New Client number 35 from ('192.168.1.247', 56968)
2025-01-06 00:43:43 - Client 35 disconnected
2025-01-06 00:43:43 - Client 35 Exit
2025-01-06 00:43:47 - New Client number 36 from ('192.168.1.247', 56990)
2025-01-06 00:44:09 - Client 36 disconnected
2025-01-06 00:44:09 - Client 36 Exit
2025-01-06 00:44:13 - New Client number 37 from ('192.168.1.247', 57008)
2025-01-06 00:44:40 - Client 37 disconnected
2025-01-06 00:44:40 - Client 37 Exit
2025-01-06 00:44:44 - New Client number 38 from ('192.168.1.247', 57032)
2025-01-06 00:45:03 - Client 38 disconnected
2025-01-06 00:45:03 - Client 38 Exit
2025-01-06 00:45:07 - New Client number 39 from ('192.168.1.247', 57050)
2025-01-06 00:45:24 - Client 39 disconnected
2025-01-06 00:45:24 - Client 39 Exit
2025-01-06 00:45:29 - New Client number 40 from ('192.168.1.247', 57066)
2025-01-06 00:47:06 - Client 40 disconnected
2025-01-06 00:47:06 - Client 40 Exit
2025-01-06 00:47:10 - New Client number 41 from ('192.168.1.247', 57088)
2025-01-06 00:48:14 - Client 41 disconnected
2025-01-06 00:48:14 - Client 41 Exit
2025-01-06 13:56:35 - Server listening on ('0.0.0.0', 3102)
2025-01-06 13:56:37 - Public server ip: 77.137.180.134, local server ip: 172.16.18.166
2025-01-06 13:56:37 - Main thread: before accepting ...
2025-01-06 13:56:42 - New Client number 1 from ('172.16.18.166', 57853)
2025-01-06 13:59:24 - New Client number 2 from ('172.16.18.166', 57890)
2025-01-06 13:59:47 - New Client number 3 from ('172.16.18.166', 57893)
2025-01-06 14:00:40 - Client 3 disconnected
2025-01-06 14:00:40 - Client 3 Exit
2025-01-06 14:00:41 - Client 2 disconnected
2025-01-06 14:00:41 - Client 2 Exit
2025-01-06 14:00:43 - Client 1 disconnected
2025-01-06 14:00:43 - Client 1 Exit
2025-01-06 17:23:23 - Server listening on ('0.0.0.0', 3102)
2025-01-06 17:23:24 - Public server ip: 46.121.5.143, local server ip: 192.168.1.247
2025-01-06 17:23:24 - Main thread: before accepting ...
2025-01-06 17:23:35 - New Client number 1 from ('192.168.1.247', 58041)
2025-01-06 17:25:39 - Client 1 disconnected
2025-01-06 17:25:39 - Client 1 Exit
2025-01-06 17:25:43 - New Client number 2 from ('192.168.1.247', 58124)
2025-01-06 17:28:33 - Client 2 disconnected
2025-01-06 17:28:33 - Client 2 Exit
2025-01-06 17:28:37 - New Client number 3 from ('192.168.1.247', 58168)
2025-01-06 17:29:15 - Client 3 disconnected
2025-01-06 17:29:15 - Client 3 Exit
2025-01-06 17:29:19 - New Client number 4 from ('192.168.1.247', 58185)
2025-01-06 17:29:37 - Client 4 disconnected
2025-01-06 17:29:37 - Client 4 Exit
2025-01-06 17:29:41 - New Client number 5 from ('192.168.1.247', 58201)
2025-01-06 17:36:56 - Client 5 disconnected
2025-01-06 17:36:56 - Client 5 Exit
2025-01-06 17:37:00 - New Client number 6 from ('192.168.1.247', 58254)
2025-01-06 17:40:09 - Client 6 disconnected
2025-01-06 17:40:09 - Client 6 Exit
2025-01-06 17:40:13 - New Client number 7 from ('192.168.1.247', 58286)
2025-01-06 17:40:48 - Client 7 disconnected
2025-01-06 17:40:48 - Client 7 Exit
2025-01-06 17:40:52 - New Client number 8 from ('192.168.1.247', 58300)
2025-01-06 17:41:28 - Client 8 disconnected
2025-01-06 17:41:28 - Client 8 Exit
2025-01-06 17:41:32 - New Client number 9 from ('192.168.1.247', 58316)
2025-01-06 19:51:27 - Client 9 disconnected
2025-01-06 19:51:27 - Client 9 Exit
2025-01-06 19:51:33 - New Client number 10 from ('192.168.1.247', 58548)
2025-01-06 19:51:46 - Client 10 disconnected
2025-01-06 19:51:46 - Client 10 Exit
2025-01-09 07:51:25 - Server listening on ('0.0.0.0', 3102)
2025-01-09 07:51:27 - Public server ip: 77.137.180.134, local server ip: 172.16.18.166
2025-01-09 07:51:28 - Main thread: before accepting ...
2025-01-09 07:51:40 - New Client number 1 from ('172.16.18.166', 63852)
2025-01-09 07:54:36 - Client 1 disconnected
2025-01-09 07:54:36 - Client 1 Exit
2025-01-09 07:54:39 - New Client number 2 from ('172.16.18.166', 63957)
2025-01-09 07:56:43 - New Client number 3 from ('172.16.18.166', 63986)
2025-01-09 07:57:07 - New Client number 4 from ('172.16.18.166', 63988)
2025-01-09 07:58:00 - Client 2 disconnected
2025-01-09 07:58:00 - Client 2 Exit
2025-01-09 07:58:01 - Client 3 disconnected
2025-01-09 07:58:01 - Client 3 Exit
2025-01-09 07:58:02 - Client 4 disconnected
2025-01-09 07:58:02 - Client 4 Exit
2025-01-09 07:58:14 - New Client number 5 from ('172.16.18.166', 64001)
2025-01-09 08:01:46 - Client 5 disconnected
2025-01-09 08:01:46 - Client 5 Exit
2025-01-09 08:01:50 - New Client number 6 from ('172.16.18.166', 64026)
2025-01-09 08:01:58 - Client 6 disconnected
2025-01-09 08:01:58 - Client 6 Exit
2025-01-09 08:43:52 - Server listening on ('0.0.0.0', 3102)
2025-01-09 08:43:53 - Public server ip: 77.137.180.134, local server ip: 172.16.18.166
2025-01-09 08:43:54 - Main thread: before accepting ...
2025-01-09 08:43:59 - New Client number 1 from ('172.16.18.166', 64138)
2025-01-09 08:50:14 - Client 1 disconnected
2025-01-09 08:50:14 - Client 1 Exit
2025-01-09 08:50:18 - New Client number 2 from ('172.16.18.166', 64202)
2025-01-09 08:53:41 - New Client number 3 from ('172.16.18.166', 64262)
2025-01-09 08:54:01 - Client 3 disconnected
2025-01-09 08:54:01 - Client 3 Exit
2025-01-09 08:57:52 - Server listening on ('0.0.0.0', 3102)
2025-01-09 08:57:54 - Public server ip: 77.137.180.134, local server ip: 172.16.18.166
2025-01-09 08:57:54 - Main thread: before accepting ...
2025-01-09 09:01:23 - New Client number 1 from ('172.16.49.189', 50399)
2025-01-09 09:03:23 - New Client number 2 from ('172.16.18.166', 64445)
2025-01-09 09:04:07 - Client 1 disconnected
2025-01-09 09:04:07 - Client 1 Exit
2025-01-09 09:04:45 - Client 2 disconnected
2025-01-09 09:04:45 - Client 2 Exit
2025-01-09 09:09:42 - New Client number 3 from ('172.16.49.202', 64830)
2025-01-09 09:10:07 - Client 3 disconnected
2025-01-09 09:10:07 - Client 3 Exit
2025-01-14 17:21:08 - Server listening on ('0.0.0.0', 3102)
2025-01-14 17:21:09 - Public server ip: 46.116.252.21, local server ip: 10.100.102.247
2025-01-14 17:21:10 - Main thread: before accepting ...
2025-01-14 17:21:16 - New Client number 1 from ('10.100.102.247', 49758)
2025-01-14 17:22:52 - Client 1 disconnected
2025-01-14 17:22:52 - Client 1 Exit
2025-01-15 10:27:32 - Server listening on ('0.0.0.0', 3102)
2025-01-15 10:27:33 - Public server ip: 77.137.180.134, local server ip: 172.16.20.119
2025-01-15 10:27:34 - Main thread: before accepting ...
2025-01-15 10:33:43 - Server listening on ('0.0.0.0', 3102)
2025-01-15 10:33:44 - Public server ip: 77.137.180.134, local server ip: 172.16.20.119
2025-01-15 10:33:46 - Main thread: before accepting ...
2025-01-15 10:33:58 - ('192.168.56.1', 60215)
2025-01-15 10:33:58 - New Client number 1 from ('172.16.20.119', 51308)
2025-01-15 10:34:23 - ('192.168.56.1', 63944)
2025-01-15 10:34:24 - New Client number 2 from ('172.16.20.119', 51311)
2025-01-15 10:34:30 - Client 2 disconnected
2025-01-15 10:34:30 - Client 2 Exit
2025-01-15 10:35:34 - Server listening on ('0.0.0.0', 3102)
2025-01-15 10:35:35 - Public server ip: 31.168.21.75, local server ip: 172.16.20.119
2025-01-15 10:35:36 - Main thread: before accepting ...
2025-01-15 10:37:22 - New DHCP request:
2025-01-15 10:37:22 - ('192.168.56.1', 56151)
2025-01-15 10:37:22 - New Client number 1 from ('172.16.20.119', 51345)
2025-01-15 10:59:26 - Server listening on ('0.0.0.0', 3102)
2025-01-15 10:59:27 - Public server ip: 77.137.180.134, local server ip: 172.16.20.119
2025-01-15 10:59:27 - Main thread: before accepting ...
2025-01-15 10:59:36 - New DHCP request:
2025-01-15 10:59:36 - ('192.168.56.1', 55980)
2025-01-15 10:59:36 - New Client number 1 from ('172.16.20.119', 51597)
2025-01-15 11:00:29 - New DHCP request:
2025-01-15 11:00:29 - ('192.168.56.1', 55981)
2025-01-15 11:00:29 - New Client number 2 from ('172.16.20.119', 51600)
2025-01-15 11:04:00 - New DHCP request:
2025-01-15 11:04:00 - ('192.168.56.1', 58622)
2025-01-15 11:04:00 - New Client number 3 from ('172.16.20.119', 51711)
2025-01-15 11:04:06 - Client 3 disconnected
2025-01-15 11:04:06 - Client 3 Exit
2025-01-15 11:04:11 - New DHCP request:
2025-01-15 11:04:11 - ('192.168.56.1', 57856)
2025-01-15 11:04:11 - New Client number 4 from ('172.16.20.119', 51720)
2025-01-15 11:04:41 - Client 2 disconnected
2025-01-15 11:04:41 - Client 2 Exit
2025-01-15 11:04:42 - Client 1 disconnected
2025-01-15 11:04:42 - Client 1 Exit
2025-01-15 11:04:45 - Client 4 disconnected
2025-01-15 11:04:45 - Client 4 Exit
2025-01-15 11:10:22 - New DHCP request:
2025-01-15 11:10:22 - ('192.168.56.1', 59999)
2025-01-15 11:10:23 - New Client number 5 from ('172.16.20.119', 51783)
2025-01-15 11:14:11 - New DHCP request:
2025-01-15 11:14:11 - ('192.168.56.1', 60729)
2025-01-15 11:14:11 - New Client number 6 from ('172.16.20.119', 51822)
2025-01-15 11:14:48 - Client 6 disconnected
2025-01-15 11:14:48 - Client 6 Exit
2025-01-15 11:16:06 - New DHCP request:
2025-01-15 11:16:06 - ('192.168.56.1', 64669)
2025-01-15 11:16:06 - New Client number 7 from ('172.16.20.119', 51825)
2025-01-15 11:16:11 - Client 7 disconnected
2025-01-15 11:16:11 - Client 7 Exit
2025-01-15 11:20:40 - New DHCP request:
2025-01-15 11:20:40 - ('192.168.56.1', 53033)
2025-01-15 11:20:40 - New Client number 8 from ('172.16.20.119', 51871)
2025-01-15 11:22:28 - New DHCP request:
2025-01-15 11:22:28 - ('192.168.56.1', 64041)
2025-01-15 11:22:28 - New Client number 9 from ('172.16.20.119', 51876)
2025-01-15 11:24:51 - New DHCP request:
2025-01-15 11:24:51 - ('172.16.20.119', 53132)
2025-01-15 11:24:51 - New Client number 10 from ('172.16.20.119', 51900)
2025-01-15 11:34:11 - Client 5 disconnected
2025-01-15 11:34:11 - Client 5 Exit
2025-01-15 11:34:12 - Client 8 disconnected
2025-01-15 11:34:12 - Client 8 Exit
2025-01-15 11:34:12 - Client 9 disconnected
2025-01-15 11:34:12 - Client 9 Exit
2025-01-15 11:34:14 - Client 10 disconnected
2025-01-15 11:34:14 - Client 10 Exit
2025-01-15 12:24:33 - Server listening on ('0.0.0.0', 3102)
2025-01-15 12:24:34 - Public server ip: 77.137.180.134, local server ip: 172.16.20.119
2025-01-15 12:24:34 - Main thread: before accepting ...
2025-01-15 12:28:16 - New DHCP request:
2025-01-15 12:28:16 - ('172.16.20.119', 60913)
2025-01-15 12:28:16 - New Client number 1 from ('172.16.20.119', 52400)
2025-01-15 12:57:27 - Server listening on ('0.0.0.0', 3102)
2025-01-15 12:57:28 - Public server ip: 77.137.180.134, local server ip: 172.16.20.119
2025-01-15 12:57:28 - Main thread: before accepting ...
2025-01-15 12:57:37 - New DHCP request:
2025-01-15 12:57:37 - ('172.16.20.119', 55878)
2025-01-15 12:57:37 - New Client number 1 from ('172.16.20.119', 52780)
2025-01-15 13:04:19 - New DHCP request:
2025-01-15 13:04:19 - ('172.16.15.194', 59496)
2025-01-15 13:04:21 - New DHCP request:
2025-01-15 13:04:21 - ('172.16.15.194', 59497)
2025-01-15 13:06:08 - Client 1 disconnected
2025-01-15 13:06:08 - Client 1 Exit
2025-01-15 13:06:15 - Server listening on ('0.0.0.0', 3102)
2025-01-15 13:06:17 - Public server ip: 77.137.180.134, local server ip: 172.16.20.119
2025-01-15 13:06:17 - Main thread: before accepting ...
2025-01-15 13:08:29 - Server listening on ('0.0.0.0', 3102)
2025-01-15 13:08:31 - Public server ip: 77.137.180.134, local server ip: 172.16.20.119
2025-01-15 13:08:32 - Main thread: before accepting ...
2025-01-15 13:08:37 - New DHCP request:
2025-01-15 13:08:37 - ('172.16.15.194', 50648)
2025-01-15 13:08:40 - New DHCP request:
2025-01-15 13:08:40 - ('172.16.15.194', 50649)
2025-01-15 13:09:17 - Server listening on ('0.0.0.0', 3102)
2025-01-15 13:09:18 - Public server ip: 77.137.180.134, local server ip: 172.16.20.119
2025-01-15 13:09:19 - Main thread: before accepting ...
2025-01-15 13:09:26 - New DHCP request:
2025-01-15 13:09:26 - ('172.16.15.194', 50651)
2025-01-15 13:09:26 - 172.16.20.119
2025-01-15 13:10:14 - Server listening on ('0.0.0.0', 3102)
2025-01-15 13:10:15 - Public server ip: 77.137.180.134, local server ip: 172.16.20.119
2025-01-15 13:10:16 - Main thread: before accepting ...
2025-01-15 13:10:22 - New DHCP request:
2025-01-15 13:10:22 - ('172.16.15.194', 50652)
2025-01-15 13:10:22 - SERR|172.16.20.119|3102
2025-01-15 13:10:25 - New DHCP request:
2025-01-15 13:10:25 - ('172.16.15.194', 50653)
2025-01-15 13:10:25 - SERR|172.16.20.119|3102
2025-01-15 13:10:42 - Server listening on ('0.0.0.0', 3102)
2025-01-15 13:10:43 - Public server ip: 77.137.180.134, local server ip: 172.16.20.119
2025-01-15 13:10:43 - Main thread: before accepting ...
2025-01-15 13:10:49 - New DHCP request:
2025-01-15 13:10:49 - ('172.16.20.119', 56851)
2025-01-15 13:10:49 - SERR|172.16.20.119|3102
2025-01-15 13:10:50 - New Client number 1 from ('172.16.20.119', 52937)
2025-01-15 13:10:54 - Client 1 disconnected
2025-01-15 13:10:54 - Client 1 Exit
2025-01-15 13:11:37 - New DHCP request:
2025-01-15 13:11:37 - ('172.16.20.119', 53535)
2025-01-15 13:11:37 - SERR|172.16.20.119|3102
2025-01-15 13:12:04 - New Client number 2 from ('172.16.20.119', 52966)
2025-01-15 13:12:48 - New DHCP request:
2025-01-15 13:12:48 - ('172.16.15.194', 56370)
2025-01-15 13:12:48 - SERR|172.16.20.119|3102
2025-01-15 13:12:49 - New Client number 3 from ('172.16.15.194', 57864)
2025-01-15 13:12:58 - Client 3 disconnected
2025-01-15 13:12:58 - Client 3 Exit
2025-01-15 13:13:18 - New DHCP request:
2025-01-15 13:13:18 - ('172.16.15.194', 56373)
2025-01-15 13:13:18 - SERR|172.16.20.119|3102
2025-01-15 13:13:18 - New Client number 4 from ('172.16.15.194', 57867)
2025-01-15 13:13:21 - Client 4 disconnected
2025-01-15 13:13:21 - Client 4 Exit
2025-01-15 13:13:25 - New DHCP request:
2025-01-15 13:13:25 - ('172.16.15.194', 56374)
2025-01-15 13:13:25 - SERR|172.16.20.119|3102
2025-01-15 13:13:27 - New Client number 5 from ('172.16.15.194', 57868)
2025-01-15 13:13:29 - Client 5 disconnected
2025-01-15 13:13:29 - Client 5 Exit
2025-01-15 13:15:01 - Server listening on ('0.0.0.0', 3102)
2025-01-15 13:15:02 - Public server ip: 77.137.180.134, local server ip: 172.16.20.119
2025-01-15 13:15:02 - Main thread: before accepting ...
2025-01-15 13:15:06 - New DHCP request:
2025-01-15 13:15:06 - ('172.16.20.119', 49665)
2025-01-15 13:15:06 - SERR|172.16.20.119|3102
2025-01-15 13:15:06 - New Client number 1 from ('172.16.20.119', 52987)
2025-01-15 13:15:08 - Client 1 disconnected
2025-01-15 13:15:08 - Client 1 Exit
2025-01-15 13:15:14 - New DHCP request:
2025-01-15 13:15:14 - ('172.16.15.194', 54091)
2025-01-15 13:15:14 - SERR|172.16.20.119|3102
2025-01-15 13:15:14 - New Client number 2 from ('172.16.15.194', 57874)
2025-01-15 13:15:51 - Client 2 disconnected
2025-01-15 13:15:51 - Client 2 Exit
2025-01-15 13:24:34 - New DHCP request:
2025-01-15 13:24:34 - ('172.16.20.119', 65075)
2025-01-15 13:24:34 - SERR|172.16.20.119|3102
2025-01-15 13:24:34 - New Client number 3 from ('172.16.20.119', 53153)
2025-01-15 13:27:01 - Server listening on ('0.0.0.0', 3102)
2025-01-15 13:27:03 - Public server ip: 77.137.180.134, local server ip: 172.16.20.119
2025-01-15 13:27:03 - Main thread: before accepting ...
2025-01-15 13:27:52 - New Client number 1 from ('172.16.20.119', 53175)
2025-01-15 14:33:40 - Traceback (most recent call last):
  File "C:\Users\idanh\Desktop\Cyber\My Projects\Final-Project\Main\server\server.py", line 946, in recv_data
    b_len += sock.recv(len_field - len(b_len))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ConnectionAbortedError: [WinError 10053] An established connection was aborted by the software in your host machine
2025-01-15 14:33:41 - Client 1 disconnected
2025-01-15 14:33:41 - Client 1 Exit
